credentials/oauth/oauth.go

```diff
@@ -42,6 +42,9 @@ func (ts TokenSource) GetRequestMetadata(ctx context.Context, uri ...string) (ma
 	if err != nil {
 		return nil, err
 	}
+	if !credentials.CheckSecurityLevel(ctx, credentials.PrivacyAndIntegrity) {
+		return nil, fmt.Errorf("connection is not secure enough to transfer TokenSource PerRPCCredentials which requires PrivacyAndIntegrity")
```

Comment:
Hmmmm.. ideally this would show more information, namely the `SecurityLevel` of the connection.

`CheckSecurityLevel` could return an `error` instead of `bool` to facilitate this.

Maybe something like `fmt.Errorf("requires SecurityLevel %v; connection has %v", requiredLevel, hasLevel)`?  And this can wrap: `fmt.Errorf("unable to transfer TokenSource PerRPCCredentials: %v", err)`?

Reply:
Good point. `CheckSecurityLevel` now returns `error` instead of `bool`, and more complete error messages are printed in `oauth.go`.